added creating themes TEDU-132 (#114)
mautic/user-documentation: docs/builders/creating_themes.rst

Creating Themes
###############

.. vale on

> :xref:`MJML` enables marketers to create and maintain beautiful and responsive Themes for Mautic easily.

Reviewer: ```suggestion
:xref:`MJML` allows marketers to create and maintain beautiful and responsive Themes for Mautic easily.
```

.. vale off

Why use email Themes?
*********************

Purpose of a Theme
==================

.. vale on

> The GrapesJs builder makes it possible to insert HTML code and edit it in a 'What You See Is What You Get' (WYSIWYG) environment.

Reviewer: ```suggestion
The GrapesJS builder makes it possible to insert HTML code and edit it in a 'What You See Is What You Get' (WYSIWYG) environment.
```

A Theme can feature a wide variety of predefined blocks and sections reflecting the desired Email design - a template from which to start.

By choosing a Theme, you can create beautiful Emails very efficiently.

.. vale off

Modifying legacy Themes
=======================

.. vale on

> The GrapesJs Builder checks the Theme configuration file before listing the available Themes, to determine which are compatible with the Builder.

Reviewer: ```suggestion
The GrapesJS Builder checks the Theme configuration file before listing the available Themes, to determine which are compatible with the Builder.
```

Since :xref:`Mautic 3` a new line defines the compatible Builders:

File name: config.json

> .. code-block:: 

Reviewer: ```suggestion
.. code-block:: json
```

    {
        "name": "Great Theme",
        "author": "Mr. Robot",
        "authorUrl": "https://mautic.org",
        "builder": "grapesjsbuilder",
        "features": [
            "email"
        ]
    }

With the builder/s defined, the Theme shows in the Theme selection page.

If you wish to support more than one Builder, specify them in an array: ``"builder": ["legacy", "grapesjsbuilder"],``

.. vale off

Creating a Theme from scratch
*****************************

.. vale on

HTML markup
===========

> It's possible to use HTML for Themes, and the GrapesJs builder offers basic WYSIWYG editing capabilities.

Reviewer: ```suggestion
It's possible to use HTML for Themes, and the GrapesJS builder offers basic WYSIWYG editing capabilities.
```

Documentation on using sections, columns, and blocks is available in the :xref:`MJML Documentation`

.. vale off

Head Components
===============

.. vale on

At present, Mautic won't process the ``<mj-head>`` tags. None of the ``<mj-attribuites>`` run, so you have to do all styling inline.

.. vale off

Body Components
===============

.. vale on

At present, Mautic processes most ``<mj-body>`` components.

**Tested elements** are: mj-button, mj-column, mj-divider, mj-image, mj-navbar, mj-section, mj-spacer, mj-text

.. vale off

Image Asset relative URLs
=========================

.. vale on

Images have to refer to the Themes folder the following way: 

``<mj-image src="{{ getAssetUrl('Themes/'~Theme~'/assets/imagename.ext', null, null, true) }}" alt="logo" align="center" width="105px" padding="10px 0"></mj-image>``

File structure

.. code-block:: 

    name.zip
    ├── assets
    │   ├── image1.ext
    │   └── image2.ext
    ├── html
    │   ├── email.html.twig
    │   ├── email.mjml.twig
    │   ├── base.html.twig
    │   └── message.html.twig
    ├── config.json
    └── thumbnail.png

Steps to save the Theme package
===============================

> Once your design in MJML is final, go through the following steps to create the Theme package:

Reviewer: ```suggestion
Once your design in MJML is finalized, go through the following steps to create the Theme package:
```

* Save your images in the Assets folder.

* Save your MJML in the ``html`` folder as ``email.mjml.twig`` AND ``email.html.twig``.

* Use the ``base.html.twig`` and ``message.html.twig`` files from the basic Theme or make your changes there.

* Save your ``config.json`` as described previously

> * Create a thumbnail - around 400px wide, 600px high.

Reviewer: ```suggestion
* Create a thumbnail - use the dimensions of 400px wide, 600px high.
```

> * Compress the contents of the folder as a Zip file

Reviewer: ```suggestion
* Compress the contents of the folder as a Zip file - ensure that the files and folders aren't within a sub-folder in the Zip file.
```

Reviewer: This is a common mistake that folks make.
